server/schedule/filter/counter.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filter
+
+import (
+	"strconv"
+)
+
+type action int
+
+const (
+	sourceFilter action = iota
+	targetFilter
+
+	ActionLen
+)
+
+var actions = [ActionLen]string{
+	"filter-source",
+	"filter-target",
+}
+
+// String implements fmt.Stringer interface.
+func (a action) String() string {
+	if a < ActionLen {
+		return actions[a]
+	}
+	return "unknown"
+}
+
+type scope int
+
+const (
+	BalanceLeader scope = iota
+	BalanceRegion
+	BalanceHotRegion
+	Label
+
+	EvictLeader
+	RegionScatter
+	ReplicaChecker
+	RuleChecker
+
+	GrantHotLeader
+	ShuffleHotRegion
+	ShuffleRegion
+	RandomMerge
+	ScopeLen
+)
+
+var scopes = [ScopeLen]string{
+	"balance-leader-scheduler",
+	"balance-region-scheduler",
+	"balance-hot-region-scheduler",
+	"label-scheduler",
+
+	"evict-leader-scheduler",
+	"region-scatter",
+	"replica-checker",
+	"rule-checker",
+
+	"grant-hot-leader-scheduler",
+	"shuffle-region-scheduler",
+	"shuffle-region-scheduler",
+	"random-merge-scheduler",
+}
+
+// String implements fmt.Stringer interface.
+func (s scope) String() string {
+	if s >= ScopeLen {
+		return "unknown"
+	}
+	return scopes[s]
+}
+
+type filterType int
+
+const (
+	excludedFilterType filterType = iota
+	storageThresholdFilterType
+	distinctScoreFilterType
+	labelConstraintFilterType
+	ruleFitFilterType
+	ruleLeaderFilterType
+	engineFilterType
+	specialUseFilterType
+	isolationFilterType
+	regionScoreFilterType
+	idFilterType
+
+	storeStateOKFilterType
+	storeStateTombstoneFilterType
+	storeStateDownFilterType
+	storeStateOfflineFilterType
+	storeStatePauseLeaderFilterType
+	storeStateSlowFilterType
+	storeStateDisconnectedFilterType
+	storeStateBusyFilterType
+	storeStateExceedRemoveLimitFilterType
+	storeStateExceedAddLimitFilterType
+	storeStateTooManySnapshotFilterType
+	storeStateTooManyPendingPeerFilterType
+	storeStateRejectLeaderFilterType
+
+	filtersLen
+)
+
+var filters = [filtersLen]string{
```

Comment:
Please add a test or use some way to make sure the sequence are same as the definition of FilterType.

Reply:
fixed by adding unit test.